service/src/main/kotlin/fi/espoo/evaka/reports/OccupancyReport.kt

```diff
@@ -459,7 +579,7 @@ WHERE sn.placement_id = ANY(:placementIds)
         val serviceNeedCoefficient = when {
             placement.familyUnitPlacement -> BigDecimal(youngChildOccupancyCoefficient)
             date < dateOfBirth.plusYears(3) -> BigDecimal(youngChildOccupancyCoefficient)
-            else -> serviceNeedCoefficients[placement.id]
+            else -> serviceNeedCoefficients[placement.placementId]
```

Comment:
I'm a bit uncertain whether a nested function inside a huge function is a good idea. Moving it outside could be worth considering even if the parameter list will get somewhat long.

Reply:
What do you think is the problem with it?

Reply:
just readability